src/dmd/parse.d

```diff
@@ -8038,6 +8058,22 @@ final class Parser(AST) : Lexer
             e = parseNewExp(null);
             break;
 
+        case TOK.ref_:
+            {
+                if (peekNext() == TOK.leftParentheses)
+                {
+                    Token* tk = peekPastParen(peek(&token));
+                    if (skipAttributes(tk, &tk) && (tk.value == TOK.goesTo || tk.value == TOK.leftCurly))
+                    {
+                        // ref (arguments) => expression
+                        // ref (arguments) { statements... }
+                        goto case_delegate;
+                    }
+                }
+                nextToken();
+                error("found `%s` when expecting function literal following %s", token.toChars(), "`ref`".ptr);
```

Comment:
Can you add a test for this error message as well?

Reply:
done. I'm not that confident about the message though.

Reply:
I think it's ok. I don't have better suggestions at the moment.